olive/passes/onnx/gptq.py

```diff
@@ -0,0 +1,149 @@
+                default_value=0,
+                description="Random seed for sampling calibration dataset. Default value is 0.",
+            ),
+            "damp_percent": PassConfigParam(
+                type_=float,
+                default_value=0.01,
+                description="Damping factor for quantization. Default value is 0.01.",
+            ),
+            "static_groups": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use static groups for quantization. Default value is False.",
+            ),
+            "true_sequential": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use true sequential for quantization. Default value is False.",
+            ),
+            "desc_act": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use descriptive activation for quantization. Default value is False.",
+            ),
+            "sym": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Symmetric quantization. Default value is True.",
+            ),
+            "gpu": PassConfigParam(
+                type_=bool,
+                default_value=True,
+                description="Use GPU for quantization. Default value is True.",
+            ),
+            "export_optimization": PassConfigParam(
+                type_=str,
+                default_value=None,
+                description="Export optimization level. Default value is None.",
+            ),
+        }
+        config.update(get_external_data_config())
+        return config
+
+    @torch.no_grad()
+    def _run_for_config(
+        self, model: PyTorchModelHandler, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> ONNXModelHandler:
+        from optimum.exporters.onnx import onnx_export_from_model
+        from transformers import AutoModelForCausalLM, AutoTokenizer, GPTQConfig
+
+        tokenizer = AutoTokenizer.from_pretrained(model.hf_config.model_name, use_fast=False)
+
+        gptq_config = GPTQConfig(
+            bits=config["bits"],
+            dataset=config["dataset"],
+            group_size=config["group_size"],
+            damp_percent=config["damp_percent"],
+            static_groups=config["static_groups"],
+            true_sequential=config["true_sequential"],
+            desc_act=config["desc_act"],
+            sym=config["sym"],
+            tokenizer=tokenizer,
+        )
+
+        def get_onnx_quant_linear(*args, **kwargs):
+            return QuantLinearORT
+
+        import auto_gptq
+
+        # GPTQ Quantization in transformers use auto_gptq under the hood, so we
+        # replace QuantLinear in auto_gptq with QuantLinearORT for quant linear layer packing
+        auto_gptq.utils.import_utils.dynamically_import_QuantLinear = get_onnx_quant_linear
```

Comment:
Perhaps a `try,finally` block would be good here like in https://github.com/microsoft/Olive/blob/9e7eba8c8d3fd500a0d0240210472c05169f17d2/olive/passes/onnx/conversion.py#L440? 
That way this function is not permanently replaced.

Reply:
We just keep a reference to the original value so that it is assigned back when our use is done.